File: solutions/python/crypto-square/1/crypto_square.py
import math
import string


def cipher_text(plain_text):
    def sanitize_text(text_to_sanitize):
        return (
            text_to_sanitize.lower()
            .strip()
            .translate(
                str.maketrans("", "", string.punctuation + string.whitespace)
            )
        )

    text = sanitize_text(plain_text)

    if not text:
        return ""

    columns = math.ceil(math.sqrt(len(text)))
    rows = math.ceil(len(text) / columns)
    text += " " * ((columns * rows) - len(text))
    return " ".join(text[i::columns] for i in range(columns))

File: solutions/go/crypto-square/1/crypto_square.go
package cryptosquare

import (
	"math"
	"regexp"
	"strings"
)

func Encode(text string) string {
	normalizedText := strings.ToLower(regexp.MustCompile(`[^a-zA-Z0-9]`).ReplaceAllString(text, ""))
	rows, columns := dimensions(len(normalizedText))
	var encoded strings.Builder
	for column := 0; column < columns; column++ {
		for row := 0; row < rows; row++ {
			index := row*columns + column
			if index < len(normalizedText) {
				encoded.WriteByte(normalizedText[index])
			} else {
				encoded.WriteString(" ")
			}
		}
		if column != columns-1 {
			encoded.WriteString(" ")
		}
	}
	return encoded.String()
}

func dimensions(length int) (rows, columns int) {
	sqrt := math.Sqrt(float64(length))
	rows = int(math.Floor(sqrt))
	columns = int(math.Ceil(sqrt))
	if rows*columns < length {
		rows++
	}
	return
}
